odb_extractor_atoms.py

```diff
@@ -0,0 +1,627 @@
+        observe_class = step['observe:class']
+        exptime = step['observe:exposureTime']
+        inst = step['instrument:instrument']
+    #     print(inst, fpuinst[inst])
+        fpu = step[fpuinst[inst]]
+        if 'instrument:filter_name' in step.keys():
+            filter_name = step["instrument:filter_name"]
+        else:
+            filter_name = 'None'
+        wavelength = step['instrument:observingWavelength']
+        if 'GMOS' in inst:
+            coadds = '1'
+            # convert wavelength to microns
+#             wavelength = '{:5.3f}'.format(float(wavelength) / 1000.)
+        else:
+            coadds = step['observe:coadds']
+        disperser = step['instrument:disperser']
+        if 'telescope:p' in step.keys():
+            p = step['telescope:p']
+        else:
+            p = '0.0'
+        if 'telescope:q' in step.keys():
+            q = step['telescope:q']
+        else:
+            q = '0.0'    
+        print('{:25} {:10} {:7} {:3} {:10} {:20} {:12} {:12} {:7} {:5} {:5}'.format(datalab, observe_class, exptime, coadds,
+                                                                       inst, fpu, filter_name, disperser, wavelength, p, q))
+        if csv and path != '':
+            print('{},{},{},{},{},{},{},{},{},{},{},{}'.format(datalab, observe_class, exptime, coadds, inst, fpu,
+                                                               filter_name, disperser, wavelength, p, q, atom), file=f)
+
+    if csv and path != '':
+        f.close()
+
+
+def seqxlsx(sequence, comment='', path=''):
+    '''Write sequence information to an Excel spreadsheet'''
+
+    obsid = sequence[0]['ocs:observationId']
+    filename = os.path.join(path, obsid + '_seq.xlsx')
+    wb = Workbook()
+    ws = wb.active
+    
+    atom = '1'
+    
+    # Comment
+    ws['A1'] = 'comment'
+    ws['B1'] = comment
+
+    # Columns
+    columns = ['datalab', 'class', 'inst', 'exptime', 'coadds', 'fpu', 'filter_name',
+               'disperser', 'wavelength', 'p', 'q', 'atom']
+    
+    row = 2
+    for ii, col in enumerate(columns):
+        _ = ws.cell(column=ii+1, row=row, value="{0}".format(col))
+    row += 1
+        
+#     print('{},{}'.format('comment', comment), file=f)
+#     print('{},{},{},{},{},{},{},{},{},{},{}'.format('datalab', 'class', 'exptime', 'coadds', 'inst', 'fpu', 
+#                                                            'disperser', 'wavelength', 'p', 'q', 'atom'), file=f)
+
+    for step in list(sequence):
+        data = []
+        data.append(step['observe:dataLabel'])
+        data.append(step['observe:class'])
+        inst = step['instrument:instrument']
+        data.append(inst)
+    #     print(inst, fpuinst[inst])
+        data.append(float(step['observe:exposureTime']))
+        if 'GMOS' in inst:
+            coadds = '1'
+            # convert wavelength to microns
+#             wavelength = '{:5.3f}'.format(float(wavelength) / 1000.)
```

Comment:
Unsure why we're formatting `wavelength` as `str` instead of `float`?

Reply:
It should be a float in the scheduler. I converted it to a string since in this case I'm just using it in a print statement.